lib/core/CJsonStateRestoreTraverser.cc

```diff
@@ -349,96 +395,133 @@ bool CJsonStateRestoreTraverser::SRapidJsonHandler::Bool(bool b) {
     return true;
 }
 
-bool CJsonStateRestoreTraverser::SRapidJsonHandler::Int(int i) {
-    s_Type = E_TokenInt;
+bool CJsonStateRestoreTraverser::SBoostJsonHandler::on_int64(std::int64_t i, std::string_view/* s*/, json::error_code&/* ec*/) {
+    s_Type = E_TokenInt64;
     if (s_RememberValue) {
         s_Value[s_NextIndex].assign(CStringUtils::typeToString(i));
     }
+
     return true;
 }
 
-bool CJsonStateRestoreTraverser::SRapidJsonHandler::Uint(unsigned u) {
-    s_Type = E_TokenUInt;
+bool CJsonStateRestoreTraverser::SBoostJsonHandler::on_uint64(std::uint64_t u, std::string_view/* s*/, json::error_code&/* ec*/) {
+    s_Type = E_TokenUInt64;
     if (s_RememberValue) {
         s_Value[s_NextIndex].assign(CStringUtils::typeToString(u));
     }
 
     return true;
 }
 
-bool CJsonStateRestoreTraverser::SRapidJsonHandler::Int64(std::int64_t i) {
-    s_Type = E_TokenInt64;
+bool CJsonStateRestoreTraverser::SBoostJsonHandler::on_double(double d, std::string_view/* s*/, json::error_code&/* ec*/) {
+    s_Type = E_TokenDouble;
     if (s_RememberValue) {
-        s_Value[s_NextIndex].assign(CStringUtils::typeToString(i));
+        s_Value[s_NextIndex].assign(CStringUtils::typeToString(d));
```

Comment:
Presumably in this case `s` already contains the string representation of the `double`, so round-tripping through an actual `double` variable is pointless. Could we just assign `s` here?

Same for other types of numbers.

According to https://www.boost.org/doc/libs/1_83_0/libs/json/doc/html/json/ref/boost__json__parse_options/numbers.html, if you use `number_precision::none` in the options then it just validates and doesn't actually parse the numbers, which should make parsing our number-heavy state _much_ faster.